### PR TITLE
on key expiration, re-send the data to re-lock the paywall

### LINKED ISSUE #4295: data iframe does not update data when a key expires
Currently, if the user is on a page when a key expires, the data iframe does not emit new data with the key being locked, so the UI cannot respond to this event. This is different from expected behavior, and different from how the legacy paywall works

### PR DESCRIPTION
**Pull request by cellog**
# Description

This PR adds missing behavior when a key expires. Currently, the paywall remains unlocked until a refresh. This PR adds a simple setTimeout call that re-dispatches the blockchain data when the key is expired. Because the "locked" state is re-calculated every time the data is sent, this will trigger a state change.

# Issues

Fixes #4295 

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by akeem** [comment]
@cellog Have you been able to manually test the old paywall and new paywall behavior with metamask and the unlock provider with this code in place? We will need to manually test changes here until we have sufficient testing in place to catch breaking changes

**Comment by cellog** [comment]
This code does not affect the old paywall. I did not yet run it on the new one

**Comment by cellog** [comment]
out

This screencast shows the data iframe sending data 1 second after key expiry, and the checkout resetting

File: paywall/src/data-iframe/blockchainHandler/BlockchainHandler.ts [hunk 1]
```diff
@@ -258,6 +283,10 @@ export default class BlockchainHandler {
     this.web3Service.on('key.updated', (_: any, key: KeyResult) => {
       key.lock = normalizeLockAddress(key.lock)
       this.store.keys[key.lock] = key
+      this.sendDataWhenKeyExpires(key.expiration)
+      // note: if this next line is ever removed, it will need to be
+      // added inside sendDataWhenKeyExpires() to ensure expired
```
Comment (julien51) [inline paywall/src/data-iframe/blockchainHandler/BlockchainHandler.ts]: It looks like this is actually called inside of `sendDataWhenKeyExpires` already?

Comment (cellog) [reply paywall/src/data-iframe/blockchainHandler/BlockchainHandler.ts]: line 232 would need to have the `dispatchChangesToPostOffice`. The flow is:

1. if the key is already expired, the `setTimeout` line is not called, and the dispatch is handled by line `290`.
2. if the key is not expired, the `setTimeout` sets up the dispatch.

The note is just to ensure the dependency on the dispatch existing in `'key.updated'` is explicit, so if in a future refactor that line is removed, we don't lose the "expired" state.

**Pull request merged**